src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -296,648 +454,673 @@ def build_process_costs(
+                    initialize=process_contingency_percentage / 100,
+                    doc="Process contingency",
+                )
 
-            self.electrical_materials_and_labor_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Electrical, materials and labor ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                # ancillary cost variables
+                self.ancillary_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.instrumentation_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.piping_materials_and_labor_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Piping, materials and labor ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.plant_services_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.electrical_materials_and_labor_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Electrical, materials and labor ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            # buildings cost variables
-            self.buildings_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Buildings cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.instrumentation_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.process_buildings_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Process buildings cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.plant_services_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
```

Comment:
Minor comment, but I dislike $MM as this is a very US-centric definition (outside the US a single M indicates millions). Also, there is no need for the currency basis to be dollars - it could well be Euros or GBP, etc.

Thus, I would recommend a more general definition of the units in the doc string.

Reply:
Addressing this comment and the one above, we don't have a dictionary of cost index values for non-USD currency units. However, users may add their own custom units by registering them, for example:

```
pyunits.load_definitions_from_strings(
        [ "custom_currency_units= 500/700 * USD_CE500",]
    )
```

where `custom_currency_units` is effectively a CE index value of 700, but really this is currency conversion rate between USD_CE500 and the new units. If users know the conversion, say 1.05 euro to 1 dollar in 2022, they could do something like

```
pyunits.load_definitions_from_strings(
        [ "EUR_2022= 1/1.05 * USD_2022",]
    )
```

We would need to update the code, variables bounds, doc strings, etc. to account for other units besides USD.

Reply:
Actually, a better answer would be to never write units into the doc strings and just store them as units on the object as this gets around the entire issue and avoids duplicating information. We should then make use of the fact that pint supports the concept of `Munits = 1e6*units` to handle the fact that this is in millions of currency units (also as a note for the future, some currencies would want different multipliers due to smaller or larger units of currency).